src/persist/src/indexed/future.rs

```diff
@@ -26,11 +26,40 @@ use crate::Data;
 /// A persistent, compacting data structure containing `(Key, Value, Time,
 /// Diff)` entries indexed by `(time, key, value)`.
 ///
+/// Future exists to hold data that has been drained out of a [crate::storage::Buffer]
+/// but not yet "seal"ed into a [crate::indexed::trace::BlobTrace]. We store incoming
+/// data as immutable batches of updates, corresponding to non-empty, sorted intervals
+/// of [crate::storage::Buffer] sequence numbers.
+///
+/// As times get sealed and the corresponding updates get moved into the trace,
+/// Future can remove those updates, and eventually, entire batches. The approach
+/// to removing sealed data optimizes for the common case where data arrives roughly
+/// in order, with a small set of distinct times represented per batch. Every
+/// future batch tracks the minimum and maximum update timestamp contained within
+/// its list of updates, and we eagerly drop batches that only contain data prior
+/// to the sealed frontier whenever possible. In the expected case where data arrives
+/// in order, most future batches only contain data for a single timestamp, and
```

Comment:
cleaned up the language to clarify that the two main assumptions are:

data arrives in order
future batches contain data for a small subrange of times

and that evictions helps when both of those hold, and otherwise, trim is the fallback.